non-normative/acpi.adoc

```diff
@@ -0,0 +1,103 @@
+[[acpi-guidance-tab-opt]]
+[cols="3,2,2", width=95%, align="center", options="header"]
+|===
+|ACPI Table                                    |ACPI Section  |Note
+|Memory-mapped Configuration space (MCFG)      |cite:[PCIFW]  |See <<acpi-guidance-pcie, the notes below>>
+| Secondary System Description Table (SSDT)    |5.2.11.2      |See <<acpi-aml>> and <<acpi-guidance-aml, the notes below>>.
+|Serial Port Console Redirection (SPCR)        |cite:[SPCR]   |See <<acpi-spcr>> and <<acpi-guidance-spcr, the notes below>>
+|System Resource Affinity Table (SRAT)         |5.2.16        |If the platform supports NUMA
+|System Locality Information Table (SLIT)      |5.2.17        |If the platform supports NUMA
+|Boot Error Record Table (BERT)                |18.3.1        |If ACPI RAS is supported
+|Error Injection Table (EINJ)                  |18.6.1        |If ACPI RAS is supported
+|Error Record Serialization Table (ERST)       |18.5          |If ACPI RAS is supported
+|Hardware Error Source Table (HEST)            |18.3.2        |If ACPI RAS is supported
+|===
+
+[[acpi-guidance-aml]]
+===== DSDT and SSDTs
+
+The ACPI name space describes devices which cannot be enumerated by any other standard ways. These typically include SoC embedded memory-mapped I/O devices, such as UARTs, <<acpi-guidance-pcie, PCIe>> or CXL root complexes, GPIO controllers, etc.
+
+It's an implementation choice if the ACPI name space is defined solely with a DSDT or with any additional SSDTs. For example, a UEFI implementation
+may choose to use SSDTs to:
+
+* describe devices that vary across SoC SKUs, revisions or variants.
+* describe devices, where the backing AML is generated or patched at boot time.
+
+// Provide guidance here for converting existing device tree node definitions to ACPI.
+
+// Provide guidance here for describing NS16550-compatible UARTs.
+
+[[acpi-guidance-fadt]]
+===== FADT
+
+cite:[ACPI] (Section 5.2.9) provides guidance on filling the
+Fixed ACPI Description Table for HW-reduced ACPI.
+
+Don't forget to select an appropriate Preferred PM Profile.
+
+[[acpi-guidance-madt]]
+===== MADT
+
+RINTC (per-hart) structures are mandatory. Depending on the interrupt controller implemented by the system, the MADT will also contain either PLIC or IMSIC/APLIC structures.
+
+[[acpi-guidance-pcie]]
+===== PCIe
+
+On some architectures, it became an industry accepted norm to describe PCIe implementations not compliant to the PCI Firmware Specification cite:[PCIFW]
+using specification-defined ACPI tables and objects. RISC-V systems compliant to the BRS must only expose ECAM-compatible implementations using the
+MCFG and the standard AML Hardware ID (HID) PNP0A08 and Compatible ID (CID) PNP0A03, and must not rely on ACPI table header information or other out-of-band
+means of detecting quirked behavior.
+
+Some minor incompatibilities, such as split PCIe configuration spaces or incorrect CFG0 filtering, can be handled by emulating ECAM accesses in privileged firmware
```

Comment:
Starting with pcifw 3.3, split PCIe configuration spaces are supported by MCFG (multiple entries for a single segment are permitted so long as the bus number ranges do not overlap).

Reply:
thanks